rest-api.md

```diff
@@ -0,0 +1,767 @@
+## API
+
+### /keys
+
+**GET** `/v1/keys`
+
+Endpoint returns list of keys for all modules.
+
+Query:
+
+- `used` - filter for used/unused keys. Possible values: true/false;
+- `operatorIndex` - filter for keys of operator with index `operatorIndex`;
+
+```typescript
+interface ELBlockSnapshot {
+  blockNumber: number;
+  blockHash: string;
+  blockTimestamp: number;
+}
+
+interface Key {
+  index: number;
+  key: string;
+  depositSignature: string;
+  used: boolean;
+  operatorIndex: number;
+  moduleAddress: string;
+}
+
+class Response {
+  data: Key[];
+  meta: {
+    elBlockSnapshot: ELBlockSnapshot;
+  };
+}
+interface HttpException {
+  statusCode: number;
+  message: string;
+}
+
+class TooEarlyResponse implements HttpException {
+  statusCode: number = 425;
+  message: string = 'Too early response';
+}
+```
+
+Example:
+
+```
+curl -X 'GET' 'http://localhost:3000/v1/keys'
+```
+
+:::warning
+Response of this endpoint could be very large but we can’t have a pagination here since data could be updated in the process.
+:::
+
+:::warning
+If API returns 425 code, it means database is not ready for work
+:::
+
+**GET** `/v1/keys/{pubkey}`
+
+Return key by public key with basic fields. `pubkey` should be in lowercase.
+
+```typescript
+class Response {
+  data: Key[];
+  meta: {
+    elBlockSnapshot: ELBlockSnapshot;
+  };
+}
+
+interface HttpException {
+  statusCode: number;
+  message: string;
+}
+
+class TooEarlyResponse implements HttpException {
+  statusCode: number = 425;
+  message: string = 'Too early response';
+}
+
+class NotFoundException implements HttpException {
+  statusCode: number = 404;
+}
+```
+
+Example:
+
+```
+curl -X 'GET' 'http://localhost:3005/v1/keys/pubkey'
+```
+
+**POST** `/v1/keys/find`
+
+Returns all keys found in db.
+
+Request body:
+
+```typescript
+interface RequestBody {
+  // public keys in lowercase
+  pubkeys: string[];
```

Comment:
request body should be just list [] or dict {'pubkeys': []}?
Why not just list of keys?

Reply:
i think it is part of implementation. for doc we dont need dict as you noticed